spec/p2p/p2p-top-level.md

```diff
@@ -0,0 +1,310 @@
+*** This is the beginning of an unfinished draft. Don't continue reading! ***
+
+# << Tendermint P2P >>
+
+<!--- > Rough outline of what the component is doing and why. 2-3 paragraphs 
+--->
+
+The Tendermint consists of multiple protocols, namely,
+- Consensus
+- Mempool
+- Evidence
+- Blocksync
+- Statesync
+
+that each plays a role in making sure that validators can produce blocks. These protocols are implemented in so-called reactors (one for each protocol) that encode two functionalities:
+
+- Protocol logic (controlling the local state of the protocols and deciding what messages to send to others, e.g., the rules we find in the arXiv paper)
+
+- Communication. Message exchange with other nodes (Gossip)
```

Comment:
I agree. I wonder whether there is another distinction within the "global" category: Consensus requires the gossip property of the paper, while for mempool and evidence it would be enough to reach a good proposer. The latter property might be harder to characterize, but would it make sense to make such a distinction clear?

Reply:
I think it is fine to be hand-waving at this point in the text. I have added a note in the "A Tendermint node" section (which me might rename). I think for each reactor, we should describe what kind of communication it needs, as this will inform what p2p needs to provide

Reply:
I agree with Lasaro here, also I consider that a higher level concept, of transmitting information could be used. Not always we learn an information from receiving a message from the original sender, we may learn (part of) the information it carried when originally sent (e.g. locally broadcast or unicast) in another way. In any case, however, we need to ensure a proper flow of information between every pair of (correct?) node.